railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb

```diff
@@ -49,7 +77,27 @@ def destroy
     @<%= orm_instance.destroy %>
     redirect_to <%= index_helper %>_url, notice: <%= "'#{human_name} was successfully destroyed.'" %>
   end
+  <% if collection_routing? %>
+  # Collection routes
+  # PATCH <%= route_url %>/1..10
+  def update_many
+    <%= orm_class.find(class_name, "#{singular_table_name}_ids") %>.each do |<%= singular_table_name %>|
+      <%= singular_table_name %>.update(params[:<%= plural_table_name %>][:"#{<%= singular_table_name %>.id}"])
+    end
+    redirect_to @<%= singular_table_name %>, notice: <%= "'#{human_name}s were successfully updated.'" %>
+  end
 
+  # PUT <%= route_url %>/1..10
+  def replace
+    <%= orm_class.destroy_all(class_name) %>
+    create
+  end
```

Comment:
We thought about the choice of using "replace" vs "replace_many" and thought that using the suffix "many" made sense when we already had a singular version of it available. Since `PUT` and `PATCH` both call `update` and collection `PATCH` calls `update_many`, `replace_many` would not really have a singular version. It's not necessarily the right call but the one that made most sense to me a year back. It's open to change :smiley: 

They way `PUT` works on collections is that it destroys the entire collection and replaces it with the payload supplied by calling `create`. But you are right, it doesn't make much sense to supply a set of ids but then that's the only way the router can differentiate between a singular and a collection request. Now in retrospect it looks like a bad design decision and it is. Any suggestions how we can fix this? Maybe we should destroy all records except the ids specified and then patch them?